### PR TITLE
Adding TauDEM

### PR DESCRIPTION
**Pull request by RyanMcCarthy-NOAA**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/taudem/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
```
Comment (scopatz) [inline recipes/taudem/meta.yaml:4]: please remove comments

File: recipes/taudem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+build:
+  number: 0
+  skip: true  # [win or osx]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - mpich
```
Comment (scopatz) [inline recipes/taudem/meta.yaml:26]: Same comments about MPI here

File: recipes/taudem/meta.yaml (hunk 1 above)
Comment (RyanMcCarthy-NOAA) [reply recipes/taudem/meta.yaml]: fixed! Thanks for the review.

File: recipes/taudem/meta.yaml (hunk 2 above)
Comment (RyanMcCarthy-NOAA) [reply recipes/taudem/meta.yaml]: Also fixed!

**Comment by RyanMcCarthy-NOAA** [comment]
@scopatz , same comments here as the other PR. The developer of TauDEM only tests with `mpich`, so I thought it safer to specify that instead of `{{ mpi }}`.

**Comment by scopatz** [comment]
Yeah, so I guess the question here is if you want to lock in everyone who uses this package to mpich?

**Comment by RyanMcCarthy-NOAA** [comment]
@scopatz , I just tested with `openmpi` (specifically adding an mpi section to `conda_build_config.yaml` as the docs specified) and TauDEM fails to build:
```
error: call to 'MPI_Type_extent' declared with attribute error: MPI_Type_extent was removed in MPI-3.0.  Use MPI_Type_get_extent instead.
```
I didn't see any 2.x versions of openmpi that I can pin to, so I think I am stuck locking people into mpich.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**